cmake/emscripten.cmake

```diff
@@ -0,0 +1,190 @@
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# Part of the Tit Solver project, under the MIT License.
+# See /LICENSE.md for license information. SPDX-License-Identifier: MIT
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+include_guard()
+include(clang)
+include(utils)
+
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+# Find the Emscripten compiler executable.
+find_program(EMCC_EXE NAMES "emcc" REQUIRED)
+
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+#
+# Add a WebAssembly library.
+#
+# WebAssembly libraries are built "from" native targets. All the source files
+# are compiled to WebAssembly, corresponding native object files are used to
+# track the dependencies (like, for example, to trigger rebuilds when the
+# source files or included headers are updated).
+#
+# WebAssembly build is limited in the following ways:
+#
+# - We do not link with any libraries, except for the standard C++ library and
+#   Emscripten's own libraries (like embind or GLFW). Therefore, only the
+#   header-only parts of the third-party libraries are available. (We may later
+#   add support for linking with our own libraries.)
+#
+# - Exceptions are disabled. Exceptions make the code significantly slower,
+#   Emscripten recommends to disable them.
+#
+function(add_wasm_library TARGET NATIVE_TARGET_OR_ALIAS)
```

Comment:
Rename to `add_wasm_binary`, since there is not much of a difference between libraries and executables in WASM world.